src/promptflow/promptflow/exceptions.py

```diff
@@ -1,10 +1,33 @@
 # ---------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
-
+import importlib
+import inspect
 import string
+import traceback
 from enum import Enum
 from functools import cached_property
+from azure.core.exceptions import HttpResponseError
+from promptflow._utils.utils import get_exception_classes
+
+
+class ErrorCategory(str, Enum):
+    SDKUserError = "SDKUserError"
+    SystemError = "SystemError"
+
+
+class ErrorType(str, Enum):
+    ExecutorError = "ExecutorError"
+    PFSError = "PFSError"
+    CoreError = "CoreError"
+    BatchError = "BatchError"
+    ContractsError = "ContractsError"
+    UtilsError = "UtilsError"
+    StorageError = "StorageError"
+    SDKError = "SDKError"
+    CLIError = "CLIError"
+    ExternalSyetemError = "ExternalSyetemError"
+    ExternalUserError = "ExternalUserError"
 
 
 class ErrorTarget(str, Enum):
```

Comment:
We can reference v2's ErrorTarget: 
![image](https://github.com/microsoft/promptflow/assets/7776147/213b4ae8-8e43-471e-a898-1114bf9bd121)

Reply:
I took a look at the "**ErrorTarget**" in V2, and it seems that the elements inside are different from the "**ErrorTarget**" in promptflow. Our "**ErrorTarget**" in promptflow already exists, and we need to add new ones based on specific business needs in the future.